datafusion/optimizer/src/extract_equijoin_predicate.rs

```diff
@@ -67,66 +88,97 @@ impl OptimizerRule for ExtractEquijoinPredicate {
             }) => {
```

Comment:
You might be able to avoid a level of indenting if you did the entire match here, like:

```
 match plan {
            LogicalPlan::Join(Join {
                left,
                right,
                mut on,
                filter: Some(expr),
                join_type,
                join_constraint,
            }) => {
```

Then you could avoid reassembling LogicalPlan::Join as well